src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -488,49 +488,55 @@ public virtual void OnDeserialization(object sender)
             }
             else
             {
-                buckets = null;
+                _buckets = null;
             }
 
-            version = realVersion;
+            _version = realVersion;
             HashHelpers.SerializationInfoTable.Remove(this);
         }
 
         private void Resize()
         {
-            Resize(HashHelpers.ExpandPrime(count), false);
+            Resize(HashHelpers.ExpandPrime(_count), false);
         }
 
         private void Resize(int newSize, bool forceNewHashCodes)
         {
-            Debug.Assert(newSize >= entries.Length);
-            int[] newBuckets = new int[newSize];
-            for (int i = 0; i < newBuckets.Length; i++) newBuckets[i] = -1;
-            Entry[] newEntries = new Entry[newSize];
-            Array.Copy(entries, 0, newEntries, 0, count);
+            Debug.Assert(newSize >= _entries.Length);
+
+            int[] buckets = new int[newSize];
+            for (int i = 0; i < buckets.Length; i++)
+            {
+                buckets[i] = -1;
+            }
+            Entry[] entries = new Entry[newSize];
+
+            int count = _count;
+            Array.Copy(_entries, 0, entries, 0, count);
 
             if (forceNewHashCodes)
             {
                 for (int i = 0; i < count; i++)
```

Comment:
e.g. with prefixes its more obvious that this is a compare to memory location than to register

G_M8319_IG11:
       FFC3                 inc      ebx
       3B5E38               cmp      ebx, dword ptr [rsi+56]
       7CC8                 jl       SHORT G_M8319_IG10